shell/initialize/entry-helpers.js

```diff
@@ -123,30 +122,29 @@ const checkForErrors = (app) => {
  */
 function callMiddleware(Components, context) {
   let midd = [];
-  let unknownMiddleware = false;
 
   Components.forEach((Component) => {
     if (Component.options.middleware) {
       midd = midd.concat(Component.options.middleware);
     }
   });
 
+  midd = midd.filter((middleware) => {
+    const isMiddlwareFunction = typeof middleware === 'function';
+
+    if (!isMiddlwareFunction) {
+      console.warn('All middleware is deprecated. For a short time inline middleware specified as a function will continue to work.', `We noticed '${ middleware }' middleware is still being used.`); // eslint-disable-line no-console
+    }
+
+    return isMiddlwareFunction;
+  });
+
   midd = midd.map((name) => {
     if (typeof name === 'function') {
       return name;
     }
```

Comment:
Once I finish removing all the inline function middleware all of this middleware code can be removed.